File: src/eu/rafig/covid/core/game/entities/effects/Effect.java
package eu.rafig.covid.core.game.entities.effects;

//TODO: if effects are more complex then redefine each effect
public class Effect {
    private String desciption;
    private boolean isActive;
    private double dnaPrice;
    private double spreading;
    private double energy;
    private double reprduction;

    public Effect(String desciption, double dnaPrice) {
        this.desciption = desciption;
        this.isActive = false;
        this.dnaPrice = dnaPrice;
    }

    public Effect(String desciption, double dnaPrice, double spreading, double energy, double reprduction) {
        this.desciption = desciption;
        this.dnaPrice = dnaPrice;
        this.spreading = spreading;
        this.energy = energy;
        this.reprduction = reprduction;
    }

    public String getDesciption() {
        return desciption;
    }

    public double getDnaPrice() {
        return dnaPrice;
    }

    public boolean isActive() {
        return isActive;
    }

    public void setActive(boolean active) {
        isActive = active;
    }

    public double getSpreading() {
        return spreading;
    }

    public double getEnergy() {
        return energy;
    }

    public double getReprduction() {
        return reprduction;
    }

}

File: Assets/src/C#/entities/events/Effect.cs
using System.Collections;
using System.Collections.Generic;
using eu.parada.common;

namespace eu.parada.entities.events {
    public class Effect {
        //TODO: if effects are more complex then redefine each effect
        public string desciption { get; private set; }
        public bool isActive { get; set; }
        public double dnaPrice { get; private set; }
        public double spreadingRate{ get; private set; }
        public double energyRate { get; private set; }
        public double reprodutctionRate { get; private set; }

        private List<string> messages;

        public Effect(string desciption, List<string> messages, double dnaPrice) {
            this.desciption = desciption;
            this.isActive = false;
            this.dnaPrice = dnaPrice;
            this.messages = messages;
        }

        public Effect (string desciption, List<string> messages, double dnaPrice, double spreading, double energy, double reproduction) {
            this.desciption = desciption;
            this.dnaPrice = dnaPrice;
            this.isActive = false;
            this.spreadingRate = spreading;
            this.energyRate = energy;
            this.reprodutctionRate = reproduction;
            this.messages = messages;
        }

        public StringMessage getRandomMessage() {
            string randomString = messages[Randomizer.getRandomNumberMax(messages.Count - 1)];
            StringMessage message = new StringMessage(randomString, desciption);
            message.setLoggable(true);
            return message;
        }

    }
}
